src/macro.ts

```diff
@@ -0,0 +1,126 @@
+// const printAST = require('ast-pretty-print')
+import type babelCore from '@babel/core'
+import {createMacro} from 'babel-plugin-macros'
+import type {MacroHandler} from 'babel-plugin-macros'
+import {getReplacement} from './helpers'
+
+const prevalMacros: MacroHandler = function prevalMacros({
+  references,
+  state,
+  babel,
+}) {
+  const fileOpts = state.file.opts
+  references.default.forEach(referencePath => {
+    const path = referencePath as babelCore.NodePath<babelCore.types.Identifier>
+    const targetPath = path.parentPath
+    if (targetPath.type === 'TaggedTemplateExpression') {
+      asTag(
+        targetPath as babelCore.NodePath<babelCore.types.TaggedTemplateExpression>,
+        fileOpts,
+        babel,
+      )
+    } else if (targetPath.type === 'CallExpression') {
+      asFunction(
+        targetPath as babelCore.NodePath<babelCore.types.CallExpression>,
+        fileOpts,
+        babel,
+      )
+    } else if (targetPath.type === 'JSXOpeningElement') {
+      const jsxElement = targetPath.parentPath
+      asJSX(
+        jsxElement as babelCore.NodePath<babelCore.types.JSXOpeningElement>,
+        fileOpts,
+        babel,
+      )
+    } else if (targetPath.type === 'JSXClosingElement') {
+      // That's okay, we already prevaled this one on its opening element.
+    } else {
+      throw new Error(
+        `babel-plugin-preval/macro can only be used as tagged template expression, function call or JSX element. You tried ${referencePath.parentPath.type}.`,
+      )
+    }
+  })
+}
+
+function asTag(
+  path: babelCore.NodePath<babelCore.types.TaggedTemplateExpression>,
+  fileOpts: babelCore.TransformOptions,
+  babel: typeof babelCore,
+) {
+  const string = path.get('quasi').evaluate().value as string
+  const replacement = getReplacement({
+    string,
+    fileOpts,
+    babel,
+  })
+
+  if (!replacement) return
```

Comment:
You may need an `// istanbul ignore next because...` comment here (see [`index.ts`](https://github.com/OpenSourceRaidGuild/babel-plugin-preval/blob/main/src/index.ts#L48)) to make the coverage checker happy.

Reply:
You did. I have added them.